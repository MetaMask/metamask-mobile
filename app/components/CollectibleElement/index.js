import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, StyleSheet, Text, View } from 'react-native';
import { colors, fontStyles } from '../../styles/common';
import { strings } from '../../../locales/i18n';
import CollectibleImage from '../CollectibleImage';

const styles = StyleSheet.create({
	itemWrapper: {
		flex: 1,
		flexDirection: 'row',
		paddingHorizontal: 15,
		paddingVertical: 10,
		borderBottomWidth: StyleSheet.hairlineWidth,
		borderBottomColor: colors.borderColor
	},
	balances: {
		flex: 1
	},
	name: {
		fontSize: 16,
		color: colors.fontPrimary,
		...fontStyles.normal
	},
	tokenId: {
		fontSize: 12,
		color: colors.fontPrimary,
		...fontStyles.normal
	}
});

/**
 * View that renders a list of Collectibles
 * also known as ERC-721 Tokens
 */
export default class CollectibleElement extends Component {
	static propTypes = {
		/**
		 * Asset object (in this case ERC721 token)
		 */
		asset: PropTypes.object
	};

	render = () => {
		const { asset } = this.props;
		return (
			<TouchableOpacity style={styles.itemWrapper} key={`asset-${asset.tokenId}`}>
				<CollectibleImage asset={asset} />
				<View style={styles.balances}>
					<Text style={styles.name}>{asset.name}</Text>
					<Text style={styles.tokenId}>
						{strings('collectible.collectible_token_id')}: {asset.tokenId}
					</Text>
				</View>
			</TouchableOpacity>
		);
	};
}
