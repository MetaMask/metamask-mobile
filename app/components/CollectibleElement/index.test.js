import React from 'react';
import { shallow } from 'enzyme';
import CollectibleElement from './';

describe('CollectibleElement', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<CollectibleElement asset={{ name: 'NAME', image: 'IMAGE', tokenId: 123, address: '0x123' }} />
		);
		expect(wrapper).toMatchSnapshot();
	});
});
