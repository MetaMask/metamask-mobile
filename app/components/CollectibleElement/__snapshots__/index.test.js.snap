// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectibleElement should render correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  key="asset-123"
  style={
    Object {
      "borderBottomColor": "#CCCCCC",
      "borderBottomWidth": 0.5,
      "flex": 1,
      "flexDirection": "row",
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <CollectibleImage
    asset={
      Object {
        "address": "0x123",
        "image": "IMAGE",
        "name": "NAME",
        "tokenId": 123,
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#333333",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      NAME
    </Component>
    <Component
      style={
        Object {
          "color": "#333333",
          "fontFamily": "Roboto",
          "fontSize": 12,
          "fontWeight": "400",
        }
      }
    >
      Collectible Identifier
      : 
      123
    </Component>
  </Component>
</TouchableOpacity>
`;
